presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingFileSystem.java

```diff
@@ -81,7 +81,7 @@ public FSDataInputStream openFile(Path path, HiveFileContext hiveFileContext)
             // FilePath is a unique identifier for a file, however it can be a long string
             // hence using md5 hash of the file path as the identifier in the cache.
             // We don't set fileId because fileId is Alluxio specific
-            FileInfo info = new FileInfo().setFileIdentifier(md5().hashString(path.toString(), UTF_8).toString())
+            FileInfo info = new FileInfo().setFileIdentifier(md5().hashString(path.toString().concat(String.valueOf(hiveFileContext.getModificationTime())), UTF_8).toString())
```

Comment:
Hi @kewang1024 @shixuan-fan 
This line will change the pattern of cache identifier, presto would see a completely cold local cache when rolling out this change.   What do you think?

Reply:
if this becomes a problem for Facebook, we can add a flag to `AlluxioCacheConfig.java`

```
@Config("cache.alluxio.invalidation-enabled")
```

and we only add `getModificationTime` when invalidation-enabled is true.
In this way FB will see no change.

What do you think @BlueStalker @beinan

Reply:
I am ok with that. And after the production test, I think cache invalidation problem has gone (without considering the growing leaked cached results), let me add the option and we can go from there.